fhoss.yaml

```diff
@@ -0,0 +1,258 @@
+    constraints:
+      - allowed_pattern: "[0-9A-Za-z+/=]+"
+        description: Must be Base64-encoded
+
+resources:
+  sig_port:
+    type: OS::Neutron::Port
+    properties:
+      network_id: { get_param: public_sig_net_id }
+      security_groups:
+        - { get_param: fhoss_sig_security_group }
+
+  mgmt_port:
+    type: OS::Neutron::Port
+    properties:
+      # Specify the network ID by string to work around OpenStack issues - see https://github.com/Metaswitch/clearwater-heat/issues/18.
+      network_id: { str_replace: { params: { x: { get_param: private_mgmt_net_id } }, template: x } }
+      security_groups:
+        - { get_param: base_mgmt_security_group }
+        - { get_param: fhoss_mgmt_security_group }
+
+  mgmt_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network_id: { get_param: public_mgmt_net_id }
+      port_id: { get_resource: mgmt_port }
+
+  server:
+    type: OS::Nova::Server
+    properties:
+      name: { str_replace: { params: {__zone__: { get_param: zone } }, template: fhoss.__zone__ } }
+      image: { get_param: image }
+      flavor: { get_param: flavor }
+      key_name: { get_param: key_name }
+      networks:
+        - port: { get_resource: mgmt_port }
+        - port: { get_resource: sig_port }
+      user_data_format: RAW
+      user_data:
+        str_replace:
+          params:
+            __zone__: { get_param: zone }
+            __public_mgmt_ip__: { get_attr: [ mgmt_floating_ip, floating_ip_address ] }
+            __private_mgmt_ip__: { get_attr: [ mgmt_port, fixed_ips, 0, ip_address ] }
+            __public_sig_ip__: { get_attr: [ sig_port, fixed_ips, 0, ip_address ] }
+            __dns_mgmt_ip__: { get_param: dns_mgmt_ip }
+            __dns_sig_ip__: { get_param: dns_sig_ip }
+            __dnssec_key__: { get_param: dnssec_key }
+          template: |
+            #!/bin/bash
+
+            # Log all output to file.
+            exec > >(tee -a /var/log/clearwater-heat-fhoss.log) 2>&1
+            set -x
+
+            # Set up eth1 for signaling
+            ip addr add __public_sig_ip__/64 dev eth1
+            ip link set dev eth1 up
+            ip route add default via fd5f:5d21:845:1402::2
+
+            # Configure the APT software source.
+            add-apt-repository ppa:rkd-u/fhoss #s
+            apt-get update #s
+
+            # Prepare to install the OpenIMS HSS noninteractively
+            # We want different signaling and management IPs. FHoSS installer
+            # doesn't deal with this and we'll need to give Tomcat the
+            # management IP afterwards as it doesn't like listening on ::.
+            #s all
+            debconf-set-selections <<< "openimscore-fhoss       openimscore-fhoss/ip_to_use     string ::"
+            debconf-set-selections <<< "openimscore-fhoss       openimscore-fhoss/hss_config    select yes"
+            # The default user should be enough for our purposes
```

Comment:
This looks wrong to me:

- the default user isn't enough - we explicitly add some more below
- the line below is commented out, but I think it'd be better to delete it